battle-engine.js

```diff
@@ -510,6 +506,16 @@ BattlePokemon = (function () {
 		}
 		return null;
 	};
+	BattlePokemon.prototype.isSuppressing = function (effectType) {
```

Comment:
This is a very good idea! I'm very impressed that you thought of something that's better than the way I would have done it.